test/typecheck/missing-ref.txt

```diff
@@ -0,0 +1,26 @@
+;;; TOOL: wat2wasm
+;;; ARGS*: --enable-function-references
+;;; ERROR: 1
+(module
+  (func $f (param (ref $first_type))
+           (result (ref $second_type))
+    (local (ref $third_type))
+  )
+)
+(;; STDERR ;;;
+out/test/typecheck/missing-ref.txt:5:24: error: undefined reference type name $first_type
+  (func $f (param (ref $first_type))
+                       ^^^^^^^^^^^
+out/test/typecheck/missing-ref.txt:6:25: error: undefined reference type name $second_type
+           (result (ref $second_type))
+                        ^^^^^^^^^^^^
+out/test/typecheck/missing-ref.txt:7:17: error: undefined reference type name $third_type
+    (local (ref $third_type))
+                ^^^^^^^^^^^
+out/test/typecheck/missing-ref.txt:5:24: error: undefined reference type name $first_type
+  (func $f (param (ref $first_type))
+                       ^^^^^^^^^^^
+out/test/typecheck/missing-ref.txt:6:25: error: undefined reference type name $second_type
```

Comment:
Just a note: the error is thrown twice. The reason is that the references are resolved as func type and as a function signature. I don't think it is a problem, but we could fix it (could be later) by removing the items of param_type_vars / result_type_vars, if somebody does not like it.